src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -1776,8 +1776,17 @@ def load_dh_private_numbers(self, numbers):
         res = self._lib.Cryptography_DH_check(dh_cdata, codes)
         self.openssl_assert(res == 1)
 
-        if codes[0] != 0:
-            raise ValueError("DH private numbers did not pass safety checks.")
+        # DH_check will return DH_NOT_SUITABLE_GENERATOR if p % 24 does not
+        # equal 11 when the generator is 2. We want to ignore that error
+        # because p % 24 == 23 is also fine. See:
+        # https://crypto.stackexchange.com/questions/12961
```

Comment:
I'd be marginally happier if this had the language "quadratic residue" and "quadratic nonresidue", but looking at the SO answer it's actually... really good?